kybra/compiler/kybra_generate/src/candid_type/func/mod.rs

```diff
@@ -58,92 +67,115 @@ impl SourceMapped<&Located<ExprKind>> {
                     },
                     name,
                     params,
-                    return_type,
+                    return_type: Box::from(ReturnType::new(return_type)),
                     mode,
                 })
             }
-            _ => return Err(crate::errors::unreachable()),
+            _ => return Err(Unreachable::error().into()),
         }
     }
 
-    fn get_func_mode(&self) -> KybraResult<Mode> {
+    fn get_func_mode(&self) -> Result<Mode, Error> {
         match &self.node {
             ExprKind::Call { args, .. } => match &args[0].node {
                 ExprKind::Subscript { value, .. } => match &value.node {
                     ExprKind::Name { id, .. } => Ok(match id.as_str() {
                         "Oneway" => Mode::Oneway,
                         "Update" => Mode::Update,
                         "Query" => Mode::Query,
-                        _ => return Err(self.return_type_mode_error()),
+                        _ => return Err(ReturnTypeMode::err_from_expr(self)),
                     }),
-                    _ => Err(self.return_type_mode_error()),
+                    _ => Err(ReturnTypeMode::err_from_expr(self)),
                 },
-                _ => Err(self.return_type_mode_error()),
+                _ => Err(ReturnTypeMode::err_from_expr(self)),
```

Comment:
Consider doing a fancy nested match so that all the things that don't match fall through to the same `_ => Err()` branch.

Also, this is probably going to be in many places so do a pass to fix all these at the same time.